### PR TITLE
Add option to generate a schema for deletion messages

### LINKED ISSUE #163: Add action to schema generation
This relates to the issue
https://github.com/sdmx-twg/sdmx-ml/issues/3

If the generated schema depends on the action, the REST API needs to be able to optionally supply an action

### PR DESCRIPTION
**Pull request by sosna**
Closes #163

### CODE REVIEW

**Comment by dosse** [comment]
@sosna If I understand correctly then this parameter is required to generate a specific data message validation schema that allows for special values (non-conform to the type defined for the component) used in delete operations?

If this is the case, do we already know which special values would be allowed to be used for this? Shouldn't we define such values first, before such schema changes and rest changes are made? 

For instance, if we use the already agreed value(s) used to indicate an "intentionally missing value" then this value is already allowed within the normal schema and no different schema would be required.

**Review by agent96 — APPROVED** [review]

**Comment by sosna** [comment]
> If I understand correctly then this parameter is required to generate a specific data message validation schema that allows for special values (non-conform to the type defined for the component) used in delete operations?

@dosse: No, this is not the reason for this PR. The DSD may define mandatory measures or attributes. However, these don't necessarily have to be present in a deletion message, depending on what you intend to delete. For example, let's say that the DSD states that, for an observation, `OBS_VALUE` (measure), `CONF_STATUS` and `OBS_STATUS` (attributes) are mandatory. If you send new observations, that's a reasonable requirement. However, if want to delete the entire observation, providing the dimension values is enough (as per the spec).

In short, if you generate a schema for a deletion message, you have to make the measures and the attributes optional. 

I hope this clarifies.

**Comment by dosse** [comment]
@sosna Very clear, thanks! All fine for me.
